nixos/modules/config/xdg/terminal-exec.nix

```
@@ -0,0 +1,58 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.xdg.terminal-exec;
+  inherit (lib) mkIf mkEnableOption mkOption mkPackageOption types;
+in
+{
+  meta.maintainers = with lib.maintainers; [ Cryolitia ];
+
+  ###### interface
+
+  options = {
+
+    xdg.terminal-exec = {
+      enable = mkEnableOption "xdg-termonal-exec, the [proposal](https://gitlab.freedesktop.org/xdg/xdg-specs/-/merge_requests/46) for XDG terminal execution utility";
+      package = mkPackageOption pkgs "xdg-terminal-exec" { };
+      settings = mkOption {
+        type = with types; attrsOf (listOf str);
+        default = { };
+        description = ''
+          Configuration options for XDG terminal execution utility.
+
+          The keys are the desktop environments, and the values are a list of terminals' [desktop file IDs](https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s02.html#desktop-file-id) to try.
+
+          To define your current desktop environment, run `echo $XDG_CURRENT_DESKTOP`.
+
+          The default value is used when the desktop environment is not found in the configuration.
+        '';
+        example = {
+          default = [ "kitty.desktop" ];
+          GNOME = [ "com.raggesilver.BlackBox.desktop" ];
```

```suggestion
          default = [ "kitty.desktop" ];
          GNOME = [ "com.raggesilver.BlackBox.desktop" "org.gnome.Terminal.desktop" ];
```

an example of multiple terminals is probably good